src/transformers/models/bart/modeling_bart.py

```diff
@@ -244,19 +252,18 @@ def forward(
                 if is_cross_attention:
                     past_key_value.is_updated[self.layer_idx] = True
 
-        attn_output, attn_weights = ALL_ATTENTION_FUNCTIONS(
-            attention_type=self.config._attn_implementation,
-            eager_attention=eager_attn_forward,
-            module=self,
-            query=query_states,
-            key=key_states,
-            value=value_states,
-            attention_mask=attention_mask,
-            training=self.training,
-            dropout=self.dropout,
+        attention_interface: Callable = ALL_ATTENTION_FUNCTIONS[self.config._attn_implementation]
```

Comment:
```suggestion
        attention_interface = eager_attn_forward
        if self.config._attn_implementation != "eager":
          attention_interface: Callable = ALL_ATTENTION_FUNCTIONS[self.config._attn_implementation]
```
let's make it explicit what is the default!

Reply:
You're too fast :D will change in a sec, wouldnt work either way with `attention_interface: Callable = ALL_ATTENTION_FUNCTIONS[self.config._attn_implementation]` since eager is not registered in the interface

Reply:
haha yep exactly